### PR TITLE
feat(TDI-44873) : cve - bump olingo odata to 4.7.1 - maintenance

### PR DESCRIPTION
**Pull request by ypiel-talend**
**What is the current behavior?** (You can also link to an open issue here)
https://jira.talendforge.org/browse/TDI-44873

**What is the new behavior?**
olingo odata bumped to 4.7.1 in tMicrosoftCrmIn/Out

**Please check if the PR fulfills these requirements**

- [x] The commit message follows Talend standard
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) ?
- [ ] The code coverage on new code >75%
- [ ] The new code does not introduce new technical issues (sonar / eslint)

**What kind of change does this PR introduce?**

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

**Review by OleksiiNimych — APPROVED** [review]
LGTM

File: main/plugins/org.talend.designer.components.localprovider/components/tMicrosoftCrmInput/tMicrosoftCrmInput_java.xml [hunk 1]
```diff
@@ -27881,24 +27881,24 @@
                 MVN="mvn:org.talend.libraries/oauth2-oidc-sdk-4.5/6.4.0"
                 REQUIRED_IF="(((AUTH_TYPE=='ONLINE') AND (API_VERSION=='API_2016_ODATA' OR API_VERSION=='API_2018_ODATA')) OR ((AUTH_TYPE=='ON_PREMISE') AND (MS_CRM_VERSION == 'CRM_2016' OR MS_CRM_VERSION == 'CRM_2018')))" />
             <IMPORT
-                NAME="odata-client-core-4.3.0-patch"
-                MODULE="odata-client-core-4.3.0-patch.jar"
-                MVN="mvn:org.talend.libraries/odata-client-core-4.3.0-patch/6.4.0"
+                NAME="odata-client-core"
+                MODULE="odata-client-core.jar"
```
Comment (jzhao-talend) [inline main/plugins/org.talend.designer.components.localprovider/components/tMicrosoftCrmInput/tMicrosoftCrmInput_java.xml:27885]: why remove the verison of the jar?

Comment (ypiel-talend) [reply main/plugins/org.talend.designer.components.localprovider/components/tMicrosoftCrmInput/tMicrosoftCrmInput_java.xml:27885]: The name of the artifact was 'odata-client-core-4.3.0-patch' as studio workaround no ? the given version was 6.4.0.
Now we want the real version of the jar whic is 4.7.1 version.
Am I wrong ?

Comment (jzhao-talend) [reply main/plugins/org.talend.designer.components.localprovider/components/tMicrosoftCrmInput/tMicrosoftCrmInput_java.xml:27885]: i think MODULE should be the jar name with version.
and yes now the GAV should be used offical one.

Comment (ypiel-talend) [reply main/plugins/org.talend.designer.components.localprovider/components/tMicrosoftCrmInput/tMicrosoftCrmInput_java.xml:27885]: I added version to MODULE nodes.

Comment (ypiel-talend) [reply main/plugins/org.talend.designer.components.localprovider/components/tMicrosoftCrmInput/tMicrosoftCrmInput_java.xml:27885]: The same for NAME attribute

**Review by OleksiiNimych — APPROVED** [review]
LGTM

**Pull request merged**